mouse/mouse.nim

```diff
@@ -1,34 +1,43 @@
-import os
-
+import std/[os, strutils, strformat]
 
 type
     Vec2D* = tuple
         x: int
         y: int
+    
+    MouseButton* = enum
+        Left, Right, Middle
+    
+    PositionKind* = enum
+        Relative, Absolute
+    
+    ScrollOrientation* = enum
+        Horizontal, Vertical
 
-proc lerp*(start: Vec2D, `end`: Vec2D, dist: float): Vec2D =
+proc lerp*(start: Vec2D, finish: Vec2D, dist: float): Vec2D =
     var
-        x = (1.0 - dist) * float(start.x) + dist * float(`end`.x)
-        y = (1.0 - dist) * float(start.y) + dist * float(`end`.y)
+        x = (1.0 - dist) * float(start.x) + dist * float(finish.x)
+        y = (1.0 - dist) * float(start.y) + dist * float(finish.y)
 
     return (int(x), int(y))
 
-proc calcRelative*(curent: Vec2D, dest: Vec2D): Vec2D = 
-    var x,y = 0
-
-    if curent.x > dest.x:
-        var x = curent.x - dest.x
-    elif curent.x < dest.x:
-        var x = dest.x - curent.x
-    elif curent.x == dest.x:
-        var x = 0
+proc calcRelative*(cur: Vec2D, dest: Vec2D): Vec2D = 
```

Comment:
In this proc you did things like:
```nim
var x = 0
if true:
  var x = 5
```

In Nim this means that you're creating a _new_ `x` variable inside of the `if true` branch and _shadowing_ the outer variable, so that it actually doesn't get changed.  I've fixed that